### PR TITLE
Calculate perplexity for interpolated model predictions correctly

### PR DESCRIPTION
**Pull request by drawlinson**
Verified it matches the original training loss when uniform and cache mass are zero. 
Small bugfix to the way perplexity is calculated may affect scores slightly (was mean-of-batch-means).

### CODE REVIEW

File: projects/rsm/rsm_experiment.py [hunk 1]
```diff
@@ -590,22 +594,42 @@ def _do_prediction(
...
+            # Alternate: torch.nn.functional.one_hot(x)
+            # labels_one_hot = torch.FloatTensor(predictor_dist_size)
+            # labels_one_hot = labels_one_hot.to(self.device)
+            # labels_one_hot.zero_()
+            # labels_one_hot.scatter_(1, pred_targets.unsqueeze(1), 1)
+            num_classes = predictor_dist_size[1]
+            labels_one_hot = torch.nn.functional.one_hot(pred_targets, num_classes=num_classes)
+            labels_one_hot = labels_one_hot.to(self.device).float()
+            #ll = (labels_one_hot * torch.log(predictor_dist)).sum(dim=[1]).mean()  This matches the training loss when other models are disabled
+            ll = (labels_one_hot * torch.log(predictor_dist)).sum(dim=[0,1])
```
Comment (onejgordon) [inline projects/rsm/rsm_experiment.py:623]: `predictor_dist` doesn't include the interpolated uniform/cache distributions since those are added into `predictions`. Should this line not be using `predictions` to give us a separate interpolated model loss?

Comment (drawlinson) [reply projects/rsm/rsm_experiment.py:623]: You're right, it should be predictions. Do you want me to make a new PR?

**Comment by onejgordon** [comment]
Great! No need, I've merged and will commit separately.
